style.json

```diff
@@ -4092,81 +4091,6 @@
         "icon-opacity": 0.4
       }
     },
-    {
-      "id": "poi-level-no-name",
-      "type": "symbol",
-      "source": "poi",
-      "source-layer": "poi",
-      "minzoom": 18,
```

Comment:
A potential regression with this change is that unnamed POIs are now rendered for "16 <= zoom < 18". Depending on their "rank" value and the collision detection algorithm, they may replace other POIs with proper names and icons that used to be visible.    
It can also make the map a bit slower, as more POIs need to be considered during the rendering (especially in dense areas).

Should we add a filter rule in the layer "poi-level-3" to exclude unnamed POIs before zoom 17 or 18 ?

Reply:
Instead of completely hiding those nameless POIs before very high zooms, what about using [`symbol-sort-key`](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-symbol-sort-key) to prioritize POIs with a name?

Concerning the map performance, I don't think it's a big deal, as nameless POIs aren't that numerous compared to name ones in dense areas.

Reply:
Yes, something like: 
```json
"symbol-sort-key": ["case",
  [["has", "name"], ["get","rank"]],
  999999
]
```
would probably work.

Or we could consider that this should be the adjusted in the tile data.
Currently for some classes, POIs with or without name may have the same weight (0.0).
See
https://github.com/Qwant/openmaptiles/blob/65e2f01965db640f1114e2a4ebb4f05856ab17a2/layers/poi/class.sql#L1-L28
and
https://github.com/Qwant/kartotherian_docker/blob/83e79eb530365cadabdc67b621724595f93ecde9/import_data/import-wikidata/wikidata_functions.sql#L52-L61

cc @remi-dupre

Reply:
I used this solution, which will also sort named POIs by rank, which I guess is an added benefit :)